crates/predictor/src/predictor/point.rs

```diff
@@ -119,32 +154,55 @@ impl Point {
 
         let percent_north = mangled_lat.ceil() - mangled_lat.floor();
         let percent_east = mangled_lon.ceil() - mangled_lon.floor();
+        let percent_down : f32 = level_down_diff / ((level_up - level_down) as f32);
```

Comment:
This is going to cause a divide by zero panic when level_up and level_down are equal